addons/hr_payroll/models/hr_salary_rule.py

```diff
@@ -165,6 +165,11 @@ class HrSalaryRule(models.Model):
     input_ids = fields.One2many('hr.rule.input', 'input_id', string='Inputs', copy=True)
     note = fields.Text(string='Description')
 
+    @api.constrains('parent_rule_id')
+    def _check_parent_rule_id(self):
+        if not self._check_recursion():
```

Comment:
This does not work, if your parent field is not `parent_id` you have to specify it as parameter

Reply:
Done @mart-e